sdk/core/azure-core/inc/azure/core/cryptography/key_encryption_key.hpp

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @file
+ * @brief A key which is used to encrypt, or wrap, another key.
+ */
+
+#pragma once
+
+#include <azure/core/context.hpp>
+
+#include <string>
+#include <vector>
+
+namespace Azure { namespace Core { namespace Cryptography {
+
+  /**
+   * @brief A key which is used to encrypt, or wrap, another key.
+   *
+   */
+  struct IKeyEncryptionKey
```

Comment:
I have seen `IFoo` used in other languages to name interfaces. Why are we doing that here in C++, for a a struct?

If we want abstract classes, use `class` and don't name it as an interface.